packages/markdown/remark/src/rehype-images.ts

```diff
@@ -3,17 +3,28 @@ import type { MarkdownVFile } from './types.js';
 
 export function rehypeImages() {
 	return () =>
-		function (tree: any, file: MarkdownVFile) {
-			visit(tree, (node) => {
-				if (node.type !== 'element') return;
-				if (node.tagName !== 'img') return;
+			function (tree: any, file: MarkdownVFile) {
+					const imageOccurrenceMap = new Map();
 
-				if (node.properties?.src) {
-					if (file.data.imagePaths?.has(node.properties.src)) {
-						node.properties['__ASTRO_IMAGE_'] = node.properties.src;
-						delete node.properties.src;
-					}
-				}
-			});
-		};
+					visit(tree, (node) => {
+							if (node.type !== 'element') return;
+							if (node.tagName !== 'img') return;
+
+							if (node.properties?.src) {
+									if (file.data.imagePaths?.has(node.properties.src)) {
+											const { alt, ...otherProps } = node.properties;
```

Comment:
You can pass the `alt` through as well, `getImage` and `Image` will handle it!

Reply:
Ah yeah good point will do